### PR TITLE
Update background_isolates to Dart 3

### PR DESCRIPTION
**Pull request by johnpryan**

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]

**Comment by domesticmouse** [comment]
Land when ci turns (mostly) green

File: background_isolate_channels/lib/main.dart [hunk 1]
```diff
@@ -78,7 +78,7 @@ class _MyHomePageState extends State<MyHomePage> {
     // [sharedPreferencesSet] does not happen before opening the
     // [SimpleDatabase] there has to be a way to refresh
     // [_SimpleDatabaseServer]'s [SharedPreferences] cached values.
-    Future.wait([sharedPreferencesSet, tempDirFuture]).then((values) {
+    [sharedPreferencesSet, tempDirFuture].wait.then((values) {
```
Comment (parlough) [inline background_isolate_channels/lib/main.dart:81]: Maybe convert this to a record so you can access the values more easily?

https://api.dart.dev/stable/3.0.0/dart-async/FutureRecord2/wait.html

Comment (johnpryan) [reply background_isolate_channels/lib/main.dart]: That's a good idea - it makes the cast and null assertion go away too.

**Comment by domesticmouse** [comment]
Le sigh

```
warning - lib/main.dart:81:43 - This API is available since SDK 3.0.0, but constraints '^3.0.0-0' don't guarantee it. Try updating the SDK constraints. - sdk_version_since
```

**Pull request closed without merging**